From e526efeb8edd2d96c4d196e53bf67136eda3fdca Mon Sep 17 00:00:00 2001
From: Utsav Agarwal <utsav.agarwal@analog.com>
Date: Fri, 8 Aug 2025 11:44:30 +0100
Subject: [PATCH] sc598-som: Disabling peripherals for faster boot

Disabling the following:

- Crypto (CRC and PKTE)
- SHARC (both cores and remoteproc)
- eMMC
- QSPI
- USB
- gptimers
- TRU
- HADC
- SPORT
- Ethernet
- SRAM
- Watchdog

This gets the kernel to boot in ~1s

Signed-off-by: Utsav Agarwal <utsav.agarwal@analog.com>
---
 arch/arm64/configs/sc598-som-ezkit_defconfig | 80 +-------------------
 1 file changed, 1 insertion(+), 79 deletions(-)

diff --git a/arch/arm64/configs/sc598-som-ezkit_defconfig b/arch/arm64/configs/sc598-som-ezkit_defconfig
index 041abcf35b61..ebb18218e5cd 100644
--- a/arch/arm64/configs/sc598-som-ezkit_defconfig
+++ b/arch/arm64/configs/sc598-som-ezkit_defconfig
@@ -1,4 +1,4 @@
-CONFIG_LOCALVERSION="-yocto-standard"
+CONFIG_LOCALVERSION="-yocto-falcon"
 CONFIG_SYSVIPC=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_AUDIT=y
@@ -97,25 +97,16 @@ CONFIG_BLK_DEV_INTEGRITY=y
 CONFIG_KSM=y
 CONFIG_MEMORY_FAILURE=y
 CONFIG_TRANSPARENT_HUGEPAGE=y
-CONFIG_NET=y
 CONFIG_PACKET=y
 CONFIG_UNIX=y
-CONFIG_INET=y
 CONFIG_IP_MULTICAST=y
 CONFIG_IP_PNP=y
 CONFIG_IP_PNP_DHCP=y
 CONFIG_IP_PNP_BOOTP=y
 # CONFIG_IPV6 is not set
-CONFIG_NETWORK_PHY_TIMESTAMPING=y
-CONFIG_NETFILTER=y
-CONFIG_NETFILTER_XT_TARGET_NFQUEUE=y
 CONFIG_IP_SET=y
 CONFIG_IP_NF_IPTABLES=y
 CONFIG_VLAN_8021Q=y
-CONFIG_NET_SCHED=y
-CONFIG_NET_SCH_CBS=y
-CONFIG_NET_SCH_ETF=y
-CONFIG_NET_SCH_MQPRIO=y
 # CONFIG_WIRELESS is not set
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
@@ -124,25 +115,14 @@ CONFIG_FW_LOADER_USER_HELPER_FALLBACK=y
 # CONFIG_ALLOW_DEV_COREDUMP is not set
 CONFIG_ARM_SCPI_PROTOCOL=y
 CONFIG_EFI_CAPSULE_LOADER=y
-CONFIG_MTD=y
-CONFIG_MTD_BLOCK=y
-CONFIG_MTD_RAW_NAND=y
-CONFIG_MTD_NAND_DENALI_DT=y
-CONFIG_MTD_SPI_NOR=y
 # CONFIG_MTD_SPI_NOR_USE_4K_SECTORS is not set
 # CONFIG_PNP_DEBUG_MESSAGES is not set
 # CONFIG_XEN_BLKDEV_FRONTEND is not set
-CONFIG_SRAM=y
 CONFIG_EEPROM_AT25=m
-CONFIG_ADI_SRAM_MMAP=y
-CONFIG_ADI_SRAM_CONTROLLER=y
-CONFIG_NETDEVICES=y
 CONFIG_MACVLAN=m
 CONFIG_MACVTAP=m
-CONFIG_NETCONSOLE=y
 CONFIG_TUN=y
 CONFIG_VETH=m
-CONFIG_VIRTIO_NET=y
 # CONFIG_NET_VENDOR_ALACRITECH is not set
 # CONFIG_NET_VENDOR_AMAZON is not set
 # CONFIG_NET_VENDOR_AMD is not set
@@ -176,7 +156,6 @@ CONFIG_VIRTIO_NET=y
 # CONFIG_NET_VENDOR_SOLARFLARE is not set
 # CONFIG_NET_VENDOR_SMSC is not set
 # CONFIG_NET_VENDOR_SOCIONEXT is not set
-CONFIG_STMMAC_ETH=y
 # CONFIG_NET_VENDOR_VIA is not set
 # CONFIG_NET_VENDOR_WIZNET is not set
 # CONFIG_NET_VENDOR_XILINX is not set
@@ -225,55 +204,11 @@ CONFIG_SYSCON_REBOOT_MODE=y
 CONFIG_SENSORS_ARM_SCPI=y
 CONFIG_CPU_THERMAL=y
 CONFIG_THERMAL_EMULATION=y
-CONFIG_WATCHDOG=y
-CONFIG_ADI_WATCHDOG=y
 CONFIG_MFD_HI6421_PMIC=y
-CONFIG_SOUND=y
-CONFIG_SND=y
-CONFIG_SND_SOC=y
-CONFIG_SND_SC5XX_PCM=y
-CONFIG_SND_SC5XX_ADAU1979=y
-CONFIG_SND_SC5XX_ADAU1962=y
-CONFIG_SND_SC5XX_SHARC_ALSA_CARD=y
-CONFIG_SND_SIMPLE_CARD=m
-CONFIG_SND_AUDIO_GRAPH_CARD=m
 # CONFIG_HID is not set
 # CONFIG_USB_HID is not set
 CONFIG_USB=y
 # CONFIG_USB_OTG is not set
-CONFIG_USB_XHCI_HCD=y
-CONFIG_USB_EHCI_HCD=y
-CONFIG_USB_EHCI_HCD_PLATFORM=y
-CONFIG_USB_OHCI_HCD=y
-CONFIG_USB_OHCI_HCD_PLATFORM=y
-CONFIG_USB_MUSB_HDRC=y
-CONFIG_USB_DWC3=y
-CONFIG_USB_DWC2=y
-CONFIG_USB_CHIPIDEA=y
-CONFIG_USB_CHIPIDEA_UDC=y
-CONFIG_USB_CHIPIDEA_HOST=y
-CONFIG_USB_ISP1760=y
-CONFIG_NOP_USB_XCEIV=y
-CONFIG_USB_ULPI=y
-CONFIG_USB_GADGET=y
-CONFIG_TYPEC=m
-CONFIG_MMC=y
-CONFIG_MMC_BLOCK_MINORS=32
-CONFIG_MMC_ARMMMCI=y
-CONFIG_MMC_SDHCI=y
-CONFIG_MMC_SDHCI_PLTFM=y
-CONFIG_MMC_SDHCI_OF_DWCMSHC=y
-CONFIG_MMC_DW=y
-CONFIG_MMC_CQHCI=y
-CONFIG_NEW_LEDS=y
-CONFIG_LEDS_CLASS=y
-CONFIG_LEDS_GPIO=y
-CONFIG_LEDS_SYSCON=y
-CONFIG_LEDS_TRIGGERS=y
-CONFIG_LEDS_TRIGGER_HEARTBEAT=y
-CONFIG_LEDS_TRIGGER_CPU=y
-CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
-CONFIG_LEDS_TRIGGER_PANIC=y
 CONFIG_RTC_CLASS=y
 CONFIG_DMADEVICES=y
 CONFIG_ADI_DMA=y
@@ -291,19 +226,6 @@ CONFIG_ARM_MHU=y
 CONFIG_PLATFORM_MHU=y
 CONFIG_ARM_SMMU=y
 CONFIG_ARM_SMMU_V3=y
-CONFIG_REMOTEPROC=y
-CONFIG_ADI_REMOTEPROC=y
-CONFIG_RPMSG_CHAR=y
-CONFIG_RPMSG_QCOM_GLINK_RPM=y
-CONFIG_RPMSG_VIRTIO=y
-CONFIG_PM_DEVFREQ=y
-CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
-CONFIG_IIO=y
-CONFIG_ADI_HADC=y
-CONFIG_ADI_ADSP_IRQ=y
-CONFIG_PHY_XGENE=y
-CONFIG_PHY_QCOM_USB_HS=y
-CONFIG_HISI_PMU=y
 CONFIG_EXT2_FS=y
 CONFIG_EXT4_FS=y
 CONFIG_EXT4_FS_POSIX_ACL=y
-- 
2.34.1

